### ISSUE TITLE
Submodule issues when using FMRIPREP workflows in your own nipype pipline

### ISSUE THREAD

**Issue by ostanley** [poldracklab/niworkflows#239]
Hello all,

Not sure if this is the correct forum to post this but I wanted to use the phase difference fieldmap workflow from FMRIPREP to prepare my fieldmaps without using the whole of FMRIPREP (I have high resolution patial brain data and therefore I do my own preprocessing). This is an issue because when I import the workflow via this command ```from fmriprep.workflows.fieldmap.phdiff import init_phdiff_wf``` and set it to run I get the following error:
``` Traceback (most recent call last):
  File "BIDSSEvsGEPhaseAnalysis.py", line 339, in <module>
    metaflow_exec_graph = metaflow.run(plugin='MultiProc', plugin_args = {'n_procs' : 14})
  File "/cfmm/home/ostanle2/.virtualenvs/nipype3.6/lib/python3.6/site-packages/nipype/pipeline/engine/workflows.py", line 592, in run
    execgraph = generate_expanded_graph(deepcopy(flatgraph))
  File "/cfmm/home/ostanle2/.virtualenvs/nipype3.6/lib/python3.6/site-packages/nipype/pipeline/engine/utils.py", line 937, in generate_expanded_graph
    graph_in = _remove_nonjoin_identity_nodes(graph_in, keep_iterables=True)
  File "/cfmm/home/ostanle2/.virtualenvs/nipype3.6/lib/python3.6/site-packages/nipype/pipeline/engine/utils.py", line 813, in _remove_nonjoin_identity_nodes
    for node in _identity_nodes(graph, not keep_iterables):
  File "/cfmm/home/ostanle2/.virtualenvs/nipype3.6/lib/python3.6/site-packages/nipype/pipeline/engine/utils.py", line 829, in _identity_nodes
    print(isinstance(node.interface, IdentityInterface))
AttributeError: 'Workflow' object has no attribute 'interface' 
```

This error can be traced back to the networkx topological_sort not exanding the fmriprep workflow since it is imported as a niworkflow.nipype object and not a nipype object. I think this is caused by the fact that the fmriprep workflow uses the niworkflow submodule of nipype and not the version I have installed on my machine.

What is the correct way to connect an FMRIPREP workflow to another nipype workflow which isn't based in niworkflows? Usually the nipype version is specified by a requirments.txt file and so both objects would use the same nipype version. Should I be modifying the FMRIPREP workflow to use the system version of nipype?

Thank you,

Olivia

**Comment by effigies** [comment]
So one possibility here is for us to move back to using nipype as a dependency, rather than a submodule. We've made a concerted effort to increase the frequency of nipype's release schedule, the lack of which was the primary problem that led to this solution, and we've now had four on-time monthly releases.

In the meantime, the only other ways to work are to either pull the fmriprep sub-workflow out and re-build it based on native nipype interfaces, or to import all of your nipype components from `niworkflows.nipype`.

@oesteban @chrisfilo How do we feel about reverting to official nipype releases? It will mean that any hot-fix niworkflows releases that happen between upstream releases need to temporarily house the entire fix in niworkflows. But I think that's a pretty small price to pay for compatibility, when we're on a <1 month cycle.

**Comment by ostanley** [comment]
Thanks for the speedy reply! For now I'll rebuild it on native nipype and I'll keep an eye on this thread so I can revert back if you guys decide to use nipype as a dependency again.

**Comment by oesteban** [comment]
I'm all in with official nipype

**Comment by chrisfilo** [comment]
+1

**Comment by effigies** [comment]
Sounds good. As there have been some fixes that we depend on since 1.0.3, I propose we switch back (#241) when 1.0.4 is released.

**Comment by effigies** [comment]
Hi @ostanley. Niworkflows 0.4.0+ interfaces will be usable with standard nipype installations, as will fMRIPrep 1.1.0+, which should be available in ~2 hours.
